src/index/DeltaTriples.cpp

```diff
@@ -158,13 +158,18 @@ void DeltaTriples::modifyTriplesImpl(CancellationHandle cancellationHandle,
   std::erase_if(triples, [&targetMap](const IdTriple<0>& triple) {
     return targetMap.contains(triple);
   });
+  // `eraseTripleInAllPermutations` does not update the block metadata.
   ql::ranges::for_each(triples, [this, &inverseMap](const IdTriple<0>& triple) {
     auto handle = inverseMap.find(triple);
     if (handle != inverseMap.end()) {
       eraseTripleInAllPermutations(handle->second);
       inverseMap.erase(triple);
     }
   });
+  // After erasing the triples update the block metadate for all permutations.
+  for (auto permutation : Permutation::ALL) {
+    locatedTriples()[static_cast<int>(permutation)].updateAugmentedMetadata();
+  }
```

Comment:
```suggestion
  // Manually update the block metadata, because `eraseTripleInAllPermutations` does not update them for performance reason.
  for (auto& permutation : locatedTriples()) {
    permutation.updateAugmentedMetadata();
  }
```
(or even `ql::ranges::for_each(locatedTriples(), &LocatedTriplesPerBlock::updateAugmentedMetadata)`.

Then you can remove the comment above in 161.